### ISSUE TITLE
Solo window freezes on attempted resize

### ISSUE THREAD

**Issue by JackyFWong** [lingtjien/Grid-Tiling-Kwin#83]
Hi. First, thanks for making the script, it has been very helpful in my productivity.

The issue: This occurs when there is one window open on one desktop. Occasionally I accidentally try to resize the window (e.g. I click the border of a Firefox window when I mean to choose a different Firefox tab). When this happens, the entire window freezes until I close and re open it.

This occurs on all applications as far as I can tell. Also, when the window is frozen, I can still interact with the application (I can see stuff happening because I have an active window title widget) through clicking or the keyboard, but the window's visuals remain frozen.

This does not when there are multiple windows on one desktop, and a window freezing on one desktop does not affect other applications on other desktops. This also does not occur when the `Grid-Tiling-Kwin` script is off.

**Comment by lingtjien** [comment]
Seems to be a problem with KWin (or my use of the API). I reduced the problem to this small example
```
var clients = workspace.clientList();
for (var i = 0; i < clients.length; i++) {
  var client = clients[i];
  if (client && client.resourceClass.toString() === 'konsole')
  {
    client.clientStepUserMovedResized.connect(function(c){
      c.geometry = {
        x: 100,
        y: 100,
        width: 200,
        height:400
      };
    });
    client.clientFinishUserMovedResized.connect(function(c)
    {
      c.geometry = {
        x: 100,
        y: 100,
        width: 200,
        height:400
      };
    });
  }
}
```
This example connects both client geometry change signals to a running `konsole` client and when you then try to resize the `konsole` client, the application no longer takes input (still takes a kill signal though). It only happens when you resize the client, moving should also trigger both these signals but it doesn't make the application unresponsive, only resizing does.

My guess is that it has something to do with both signals setting the geometry property simultaneously when you release your cursor, because if you uncomment the first signal `clientStepUserMovedResized` then it won't cause a hang anymore, but you also don't get the gradual client resizing only after you release the cursor (which is actually better performance wise, since it doesn't have to resize all clients continuously, only at the end when the user releases the mouse)

**Comment by Crendgrim** [comment]
I have encountered this behaviour as well when starting a resize from the screen edges. Triggering a new geometry by adding/removing other clients to/from the same screen also unfreezes the broken window. Sometimes simply swapping the window with another helps as well, though I have found this to not be completely reliable.

**Comment by lingtjien** [comment]
submitted https://bugs.kde.org/show_bug.cgi?id=426988

**Comment by lingtjien** [comment]
that's fast https://invent.kde.org/plasma/kwin/-/merge_requests/298

**Comment by lingtjien** [comment]
Well, they shipped it (already merged to trunk) and the report says fixed in 5.20

**Comment by lingtjien** [comment]
Well, I got it in my Arch packages update today and it seems to work now 😄
